### PR TITLE
Add mock setup

### PR DESCRIPTION
**Pull request by Vimalakirti**
This PR added the feature of mock setup, which allows us to benchmark proof generation without the needs to wait for real setup.

### CODE REVIEW

File: src/bin/mock_prove.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  inputs: &Vec<&ArrayD<Fr>>,
+  graph: &Graph,
+  models: &Vec<&ArrayD<Fr>>,
+  timing: &mut TimingTree,
+) -> Result<Vec<Vec<ArrayD<Fr>>>, util::CQOutOfRangeError> {
+  // Run:
+  timed!(timing, "run witness generation", graph.run(inputs, models))
+}
+
+fn prove(
```
Comment (liliatangxy) [inline src/bin/mock_prove.rs]: is there a way to reuse the run/prove/main code from the other main?

Comment (Vimalakirti) [reply src/bin/mock_prove.rs]: yeah, we may put them under `util/`.
Please note that the `prove` in this file does not load setup from file as the `prove` in `main.rs` does, which means we will need to modify the `prove` in both files to make an unified API for them. Do you think this proposal make sense to you?

Comment (liliatangxy) [reply src/bin/mock_prove.rs]: right, i think we could maybe have a flag for mock vs. real and condition deserializing from file based on that

**Review by liliatangxy — DISMISSED** [review]

File: src/basic_block.rs [hunk 2]
```diff
@@ -169,6 +169,11 @@ pub trait BasicBlock: std::fmt::Debug + Send + Sync {
     (Vec::new(), Vec::new(), Vec::new())
   }
 
+  fn mockSetup(&self, _srs: &SRS, _model: &ArrayD<Data>) -> (Vec<G1Projective>, Vec<G2Projective>, Vec<DensePolynomial<Fr>>) {
```
Comment (ddkang) [inline src/basic_block.rs]: Is there a way to just infer this directly from the type signature? This seems pretty inelegant

Comment (Vimalakirti) [reply src/basic_block.rs]: yeah, the new commit addressed this issue.

**Review by ddkang — APPROVED** [review]

**Review by liliatangxy — APPROVED** [review]

**Pull request merged**